Unable to build libxml2 in latest macos

Unable to build libxml2

Repro: nix-env -i libxml2

make[4]: Entering directory '/private/var/folders/7g/1ptqqd597r76dqzk7hjy0bgc0000gn/T/nix-build-libxml2-2.9.10.drv-0/libxml2-2.9.10/python'
  CC       libxml.lo
  CC       libxml2-py.lo
  CC       types.lo
libxml.c:1649:43: warning: passing 'unsigned char *' to parameter of type 'const char *' converts between pointers to integer types with different sign [-Wpointer-sign]
        message = libxml_charPtrConstWrap(ptr);
                                          ^~~
./libxml_wrap.h:240:48: note: passing argument to parameter 'str' here
PyObject * libxml_charPtrConstWrap(const char *str);
                                               ^
1 warning generated.
  CCLD     libxml2mod.la
ld: library not found for -lintl
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [Makefile:600: libxml2mod.la] Error 1
make[4]: Leaving directory '/private/var/folders/7g/1ptqqd597r76dqzk7hjy0bgc0000gn/T/nix-build-libxml2-2.9.10.drv-0/libxml2-2.9.10/python'
make[3]: *** [Makefile:694: all-recursive] Error 1
make[3]: Leaving directory '/private/var/folders/7g/1ptqqd597r76dqzk7hjy0bgc0000gn/T/nix-build-libxml2-2.9.10.drv-0/libxml2-2.9.10/python'
make[2]: *** [Makefile:528: all] Error 2
make[2]: Leaving directory '/private/var/folders/7g/1ptqqd597r76dqzk7hjy0bgc0000gn/T/nix-build-libxml2-2.9.10.drv-0/libxml2-2.9.10/python'
make[1]: *** [Makefile:1479: all-recursive] Error 1
make[1]: Leaving directory '/private/var/folders/7g/1ptqqd597r76dqzk7hjy0bgc0000gn/T/nix-build-libxml2-2.9.10.drv-0/libxml2-2.9.10'
make: *** [Makefile:893: all] Error 2
builder for '/nix/store/xcqdl8cfjjbjvh7g9ka0ip3r4awy80a8-libxml2-2.9.10.drv' failed with exit code 2
error: build of '/nix/store/xcqdl8cfjjbjvh7g9ka0ip3r4awy80a8-libxml2-2.9.10.drv' failed

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.5.0, macOS 10.15.5`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.6`
 - channels(swarnpriya): `"nixpkgs-20.09pre230446.a84cbb60f02"`
 - nixpkgs: `/Users/swarnpriya/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/libraries/libxml2/default.nix
+++ b/pkgs/development/libraries/libxml2/default.nix
@@ -1,5 +1,5 @@
 { stdenv, lib, fetchurl, fetchpatch
-, zlib, xz, python, ncurses, findXMLCatalogs
+, zlib, xz, python, gettext, ncurses, findXMLCatalogs
 , pythonSupport ? stdenv.buildPlatform == stdenv.hostPlatform
 , icuSupport ? false, icu ? null
 , enableShared ? stdenv.hostPlatform.libc != "msvcrt"
@@ -44,6 +44,7 @@ stdenv.mkDerivation rec {
     ++ lib.optional (enableStatic && enableShared) "static";
 
   buildInputs = lib.optional pythonSupport python
+    ++ lib.optional (pythonSupport && python?isPy2 && python.isPy2) gettext
     ++ lib.optional (pythonSupport && python?isPy3 && python.isPy3) ncurses
     # Libxml2 has an optional dependency on liblzma.  However, on impure
     # platforms, it may end up using that from /usr/lib, and thus lack a

